src/stream/src/common/table/state_table.rs

```diff
@@ -596,8 +555,11 @@ impl<S: StateStore> StateTable<S> {
 }
 // write
 impl<S: StateStore> StateTable<S> {
-    #[expect(clippy::boxed_local)]
-    fn handle_mem_table_error(&self, e: Box<MemTableError>) {
+    fn handle_mem_table_error(&self, e: StorageError) {
+        let e = match e {
+            StorageError::MemTable(e) => e,
+            _ => unreachable!("should only get memtable error"),
+        };
```

Comment:
strange, We should change the return error type of the `delete` and `insert` function, or handle all kinds of storage error here.

Reply:
Since `delete` and `insert` will be one of the method of local state stores, I think we'd better keep the error type consistent with other methods. 

The `handle_mem_table_error` method will panic at any error. The semantic of this method can be seen as matching out the `MemtableError` for deserializing the key and value into row and print the row out, and then panic anyway.